conda_meta/meta.yaml

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+{% set version = environ.get('GIT_DESCRIBE_TAG', '0.0.0.dev').lstrip('v') %}
+{% set minor_version = version.split('.')[0] + '.' + version.split('.')[1] %}
+{% set cuda_version = '.'.join(environ['RAPIDS_CUDA_VERSION'].split('.')[:2]) %}
+{% set cuda_major = cuda_version.split('.')[0] %}
+{% set cuda_spec = ">=" + cuda_major ~ ",<" + (cuda_major | int + 1) ~ ".0a0" %} # i.e. >=11,<12.0a0
+{% set date_string = environ['RAPIDS_DATE_STRING'] %}
+
+package:
+  name: rapids_dependencies
+
+outputs:
+  - name: rapids_cupy_dependency
+    version: {{ version }}
+    requirements:
+      run:
+        - cupy >=12
+  - name: rapids_numpy_dependency
+    requirements:
+      run:
+        - numpy >=1.21
+  - name: rapids_numba_dependency
+    requirements:
+      run:
+        - numba >=0.56.4,<0.57
+  - name: rapids_cuda_python_dependency
+    requirements:
+      host:
+        - cuda-version == {{ cuda_version }}
+      run: 
+        # Note that this will require building for multiple CUDA versions
+        - pin_compatible("cuda-version", min_pin="x", max_pin="x")
+        {% if cuda_version.split('.')[0] == 1 %}
+        - cuda-python >=11.7.1,<12
+        {% else %}
+        - cuda-python >=12.0.0
+        {% endif %}
+  - name: rapids_cmake_dependency
+    requirements:
+      run:
+        - cmake>=3.23.1,!=3.25.0
+  - name: rapids_cython_dependency
+    requirements:
+      run:
+        - cython >=0.29,<0.30
+  - name: rapids_scikit_build_dependency
+    requirements:
+      run:
+        - scikit-build>=0.13.1,<0.17.2
+  # TODO: Should the grouped metapackages be named *_dependencies instead of *_dependency?
+  - name: rapids_build_dependency
```

Comment:
This feels a lot like something we've been trying to get rid of: https://github.com/rapidsai/integration/blob/branch-23.06/conda/recipes/rapids-build-env/meta.yaml

Reply:
I see where that's coming from, but there are important differences to consider:
- The rapids-build-env was used to construct the base environment in which we did things like builds and run tests. It was deeply baked into our workflows in a way that did not really allow for working around. The metapackages proposed in this PR are much simpler and are consumed by dependency lists in package specs, which is a much more self-contained usage.
- The rapids-build-env was essentially required to contain all dependencies for all packages, leading to the bloat that you mention. In this instance the packages will be much smaller and more surgical.
- Any package that needs to can simply drop usage of the metapackage with no real consequences (other than potentially going out of sync with the rest of RAPIDS).